stdlib/public/core/TypeLayoutConstraints.swift

```diff
@@ -0,0 +1,40 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// Define some compile-known protocols used to constraint type layout.
+// These protocols can be used e.g. in @_specialize attribute.
+
+// Type should be a trivial type. It's size should be 8 bits.
+public protocol _Trivial8 {
```

Comment:
@slavapestov If we do not represent these constraints internally as protocols, what would you use to represent them? Are you saying we should not represent them as types at all, but e.g. as special attributes? Or do you suggest to represent them as special (internal) types? Can you elaborate?